crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -2924,7 +2924,18 @@ impl<'db> TypeInferenceBuilder<'db> {
             | Type::TypeVar(..)
             | Type::AlwaysTruthy
             | Type::AlwaysFalsy => {
-                match object_ty.class_member(db, attribute.into()) {
+                let dataclass_params = match object_ty {
+                    Type::NominalInstance(instance) => match instance.class() {
+                        ClassType::NonGeneric(cls) => cls.dataclass_params(self.db()),
+                        ClassType::Generic(cls) => {
+                            cls.origin(self.db()).dataclass_params(self.db())
+                        }
+                    },
+                    _ => None,
+                };
+                let ro =
```

Comment:
This is a bit too cryptic for our usual coding style, we could call it `read_only`?